runtime/jvmti/jvmtiStartup.c

```diff
@@ -162,15 +166,28 @@ createAgentLibraryWithOption(J9JavaVM *vm, J9VMInitArgs *argsList, IDATA agentIn
 			UDATA optionsLength = 0;
 
 			parseLibraryAndOptions(optionsPtr, &libraryLength, &options, &optionsLength);
-			result = createAgentLibrary(vm, optionsPtr, libraryLength, options, optionsLength, TRUE, agentLibrary);
+			result = createAgentLibrary(vm, optionsPtr, libraryLength, options, optionsLength, decorate, agentLibrary);
 			Trc_JVMTI_createAgentLibraryWithOption_agentlib_result(optionsPtr, libraryLength, options, optionsLength, *agentLibrary, result);
 		}
 #if defined(J9VM_OPT_CRIU_SUPPORT)
-		if ((JNI_OK == result)
-			&& (isXrunjdwp || (0 == strncmp(JDWP_AGENT, optionsPtr, libraryLength)))
-			&& (vm->internalVMFunctions->isDebugOnRestoreEnabled(vm))
-		) {
-			*isJDWPagent = TRUE;
+		if ((JNI_OK == result) && vm->internalVMFunctions->isDebugOnRestoreEnabled(vm)) {
+			/* 1. If isXrunjdwp is TRUE, the agent option is MAPOPT_XRUNJDWP which implies a JDWP agent.
+			 * 2. If decorate is TRUE, the agent option is VMOPT_AGENTLIB_COLON,
+			 *    just compare the platform independent library name.
+			 * 3. If decorate is FALSE, the agent option is VMOPT_AGENTPATH_COLON,
+			 *    compare the actual platform-specific library name with libjdwp.so.
+			 *    CRIU only supports Linux OS flavours.
+			 */
+#define LIB_JDWP "libjdwp.so"
```

Comment:
This name is platform-specific. That should be explicit here:
```c
#if defined(LINUX)
#define LIB_JDWP "lib" JDWP_AGENT ".so"
#else /* defined(LINUX) */
#error "CRIU is only supported on Linux"
#endif /* defined(LINUX) */
```

Reply:
Added platform define.